libcst/_parser/tests/test_node_identity.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from collections import Counter
+from textwrap import dedent
+
+import libcst as cst
+from libcst.testing.utils import UnitTest, data_provider
+
+
+class DuplicateLeafNodeTest(UnitTest):
+    @data_provider(
+        (
+            # Simple program
+            (
+                """
+                foo = 'toplevel'
+                fn1(foo)
+                fn2(foo)
+                def fn_def():
+                    foo = 'shadow'
+                    fn3(foo)
+            """,
+            ),
+        )
+    )
+    def test_tokenize(self, code: str) -> None:
+        class CountVisitor(cst.CSTVisitor):
+            def __init__(self) -> None:
+                self.count = Counter()
+                self.nodes = {}
+
+            def on_visit(self, node: cst.CSTNode) -> bool:
+                self.count[id(node)] += 1
+                if self.count[id(node)] > 1:
+                    raise Exception(
```

Comment:
Since this is a test, can we use the standard test assert: `assertEqual(self.count[id(node)], 1, f"Node duplication detected between {node} and {self.nodes[id(node)]}")`?

Reply:
Good idea.